code/script.js

```diff
@@ -222,104 +234,133 @@ const generateBoard = () => {
   if (category === 'hair' || category === 'eyes') {
-
+    if (secret[category] === value) {
+      filterCharacters(true)
+    } else {
+      filterCharacters()
+    }
   } else if (category === 'accessories' || category === 'other') {
-
+    if (secret[category].includes(value)) {
+      filterCharacters(true)
+    } else {
+      filterCharacters()
+    }
   }
 }
 
-// It'll filter the characters array and redraw the game board.
+// Filter out characters via keep/ or not.
 const filterCharacters = (keep) => {
   const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
+  // Show the correct alert message depending on what we´re filtering out
   if (category === 'accessories') {
     if (keep) {
-      alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
+      alert(`Yes, the character is looking fabulous in ${value}! Confirm to keep all people that wears ${value}`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
     } else {
-      alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
+      alert(`No, this character can't pull of ${value}! Remove everyone wearing ${value}!`)
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
     }
   } else if (category === 'other') {
-    // Similar to the one above
+    if (keep) {
+      alert(`Yes some ${value}s coming up! No need to see the other ones!`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+    } else {
+      alert(`Nope, no ${value}s here! Get rid of them!`);
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+    }
   } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(`Yay, this character has beautiful ${value} ${category}. Sort out the other ones that don't!`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(`Nope this character doesn't have ${value} ${category}! Sort out all the characters that do!`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  //Adding a guess for each time player makes a guess 
+  guesses++
```

Comment:
In the guesses part maybe is a good idea to have a response after certain amount of guesses ? like after 3 guesses: you loss so it adds a little bit of tension to the game hehe 😈